### PR TITLE
Homework 1

### PR DESCRIPTION
**Pull request by Grohot9**

### CODE REVIEW

File: homework_1/decorators/cache_decorator.py [hunk 1]
```diff
@@ -1,9 +1,9 @@
 import functools
-import requests
 
 
 def lfu_cache(max_limit=64):
-    """Decorator for caching with deleting elements by LFU algorithm after going beyond the limit"""
+    """A decorator for caching with deleting elements by LFU algorithm after going beyond the limit"""
+
     def internal(func):
         @functools.wraps(func)
```
Comment (mikolaz27) [inline homework_1/decorators/cache_decorator.py:8]: remove comments, it's weird to comment on each line.
And also dont use russian lanuage comments, only English.

File: homework_1/decorators/cache_decorator.py [hunk 2]
```diff
@@ -48,10 +48,3 @@ def wrapper(*args, **kwargs):
         return wrapper
```
Comment (mikolaz27) [inline homework_1/decorators/cache_decorator.py:30]: Remove for statement. You dont need it.  Extend wrapper._cache to store count of calls.

File: homework_1/decorators/memory_decorator.py [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+def memory(func):
+    """A decorator that measures how much a function occupies in the non-swapped physical memory"""
+
+    @functools.wraps(func)
+    def wrapper(*args, **kwargs):
+        process = psutil.Process()
+        memory_usage_before = process.memory_info().rss
+        result = func(*args, **kwargs)
+        memory_usage_after = process.memory_info().rss
+        print(f"The function took up {memory_usage_after - memory_usage_before} Byte of the physical memory.")
```
Comment (mikolaz27) [inline homework_1/decorators/memory_decorator.py]: change to MB

File: homework_1/decorators/cache_decorator.py (hunk 2 above)
Comment (Grohot9) [reply homework_1/decorators/cache_decorator.py:48]: I guess you meant that I need to delete the "return cache_data". If I'll remove the "return wrapper" function will collapse.
wrapper._cache is already extended and already stores the number of calls. Now there is a nested dictionary, but I will redo it by the list in the dictionary.

File: homework_1/decorators/cache_decorator.py (hunk 1 above)
Comment (Grohot9) [reply homework_1/decorators/cache_decorator.py:8]: ✅

File: homework_1/decorators/memory_decorator.py (hunk 3 above)
Comment (Grohot9) [reply homework_1/decorators/memory_decorator.py]: I redid this task with tracemalloc lib. and changed the size of used memory from Bytes to MB in the output

File: homework_1/decorators/cache_decorator.py (hunk 2 above)
Comment (Grohot9) [reply homework_1/decorators/cache_decorator.py:30]: I guess I did what you are talking about

**Review by mikolaz27 — APPROVED** [review]

**Pull request merged**